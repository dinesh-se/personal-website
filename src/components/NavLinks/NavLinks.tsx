import clsx from 'clsx';
import Link from 'next/link';
import { usePathname } from 'next/navigation';

import { NavLinks as NavLinksType } from '@types';

const NavLinks = ({
	links,
	linkActiveState,
	linkDefaultState,
	otherStyleClasses,
}: NavLinksType) => {
	const pathname = usePathname();

	return (
		<>
			{links.map(({ label, href }) => (
				<Link
					key={href}
					href={href}
					className={clsx(
						otherStyleClasses,
						pathname === href ? linkActiveState : linkDefaultState
					)}
				>
					{label}
				</Link>
			))}
		</>
	);
};

export default NavLinks;
