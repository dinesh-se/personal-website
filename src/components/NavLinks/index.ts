'use client';

export { default as NavLinks } from './NavLinks';
