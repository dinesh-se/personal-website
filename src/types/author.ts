export interface Author {
  profile: Profile;
}

interface Profile {
  fullName: string;
  interests: string[];
  summary: string;
  contactDetail: ContactDetail;
  experience: Experience;
  testWyswyg: object;
}

interface ContactDetail {
  eMail: string;
  mobileNumber: string[];
  socialMedia: SocialMedia;
}

interface SocialMedia {
  linkedin: string;
  github: string;
}

export interface Contact extends SocialMedia, Pick<ContactDetail, 'eMail'> {}

export interface Experience {
  organizations: Organization[];
}

interface Organization {
  orgName: string;
  title: string;
  from: string;
  to: string;
  orgLogo: OrgLogo;
}

interface OrgLogo {
  url: string;
}
