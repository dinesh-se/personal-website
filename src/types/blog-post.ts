export interface BlogPost {
	id: string;
	title: string;
	description: string;
	published_at: string;
	url: string;
	comments_count: number;
	public_reactions_count: number;
	page_views_count: number;
}

export interface BlogPostUI
	extends Pick<BlogPost, 'title' | 'description' | 'url'> {
	id?: string;
	date: Date;
	commentsCount: number;
	reactionsCount: number;
	pageViewsCount: number;
}
