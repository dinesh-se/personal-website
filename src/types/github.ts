export interface Repos {
  data: Repo[];
};

export interface Repo {
  node_id: string;
  name: string;
  description: string;
  html_url: string;
  language: string;
};

export interface RepoUI extends Omit<Repo, 'node_id' | 'html_url'> {
  href: Repo['html_url']
};

export interface QueryParams {
  per_page?: string;
  sort?: string;
  since?: string
};
