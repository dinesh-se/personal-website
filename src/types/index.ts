export * from './nav-links';
export * from './author';
