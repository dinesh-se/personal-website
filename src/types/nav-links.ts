export interface NavLinks {
	links: Link[];
	linkActiveState: string;
	linkDefaultState?: string;
	otherStyleClasses?: string;
}

export interface Link {
	label: string;
	href: string;
}
