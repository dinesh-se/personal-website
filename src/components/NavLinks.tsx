'use client'

import Link from 'next/link';
import clsx from 'clsx';
import { usePathname } from 'next/navigation';

import { NavLinks } from '../types/nav-links';

const NavLinks = ({
  links,
  linkActiveState,
  linkDefaultState,
  otherStyleClasses
}: NavLinks) => {
  const pathname = usePathname();

  return (
    <>
      {links.map(({label, href}) => (
        <Link 
          key={href}
          href={href} 
          className={clsx(otherStyleClasses, pathname === href ? linkActiveState : linkDefaultState)}>
          {label}
        </Link>
      ))}
    </>
  )
};

export default NavLinks;
