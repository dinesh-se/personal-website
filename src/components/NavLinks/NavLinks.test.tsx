import { render, screen } from '@testing-library/react';

import { Link, NavLinks as NavLinksType } from '@root/src/types';

import NavLinks from './NavLinks';

const mockUsePathname = jest.fn();

jest.mock('next/navigation', () => ({
	usePathname() {
		return mockUsePathname();
	},
}));

describe('NavLinks', () => {
	it('renders correctly with a active class', () => {
		const links: Link[] = [
			{
				label: 'label-1',
				href: '/label-1',
			},
			{
				label: 'label-2',
				href: '/label-2',
			},
		];
		const props: NavLinksType = {
			links,
			linkActiveState: 'active-class',
			linkDefaultState: 'default-class',
			otherStyleClasses: 'other-class',
		};
		mockUsePathname.mockImplementation(() => '/label-1');
		const { container } = render(<NavLinks {...props} />);

		expect(container).toMatchSnapshot();
		expect(screen.getByText('label-1')).toHaveClass('active-class');
		expect(screen.getByText('label-2')).not.toHaveClass('active-class');
	});

	it('renders correctly with no active class', () => {
		const links: Link[] = [
			{
				label: 'label-1',
				href: '/label-1',
			},
			{
				label: 'label-2',
				href: '/label-2',
			},
		];
		const props: NavLinksType = {
			links,
			linkActiveState: 'active-class',
			linkDefaultState: 'default-class',
			otherStyleClasses: 'other-class',
		};
		mockUsePathname.mockImplementation(() => '/label-unknown');
		const { container } = render(<NavLinks {...props} />);

		expect(container).toMatchSnapshot();
		expect(screen.getByText('label-1')).not.toHaveClass('active-class');
		expect(screen.getByText('label-2')).not.toHaveClass('active-class');
	});
});
