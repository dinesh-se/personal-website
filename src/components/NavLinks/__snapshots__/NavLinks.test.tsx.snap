// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders correctly with a active class 1`] = `
<div>
  <a
    class="other-class active-class"
    href="/label-1"
  >
    label-1
  </a>
  <a
    class="other-class default-class"
    href="/label-2"
  >
    label-2
  </a>
</div>
`;

exports[`NavLinks renders correctly with no active class 1`] = `
<div>
  <a
    class="other-class default-class"
    href="/label-1"
  >
    label-1
  </a>
  <a
    class="other-class default-class"
    href="/label-2"
  >
    label-2
  </a>
</div>
`;
